pkg/daemon/gateway_linux.go

```diff
@@ -415,6 +415,104 @@ func (c *Controller) deletePolicyRouting(family int, _ string, priority, tableID
 	return nil
 }
 
+// findRulePositions locates all occurrences of a rule in the specified table and chain,
+// searching from lowest to highest priority (bottom to top).
+// Returns all matching position indices, or an empty slice if none found.
+func findRulePositions(ipt *iptables.IPTables, rule util.IPTableRule) ([]int, error) {
+	rules, err := ipt.List(rule.Table, rule.Chain)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list iptables rules: %w", err)
+	}
+
+	positions := make([]int, 0)
+
+	// Start from the end of the list (lowest priority) and go up to 1 (but skip index 0)
```

Comment:
Why skip 0 here?

Reply:
The rule at index zero would be the policy rule. This was also omitted in the original code snippet. I just follow the spirit here.